diff --git a/vcpkg-port-config.cmake b/vcpkg-port-config.cmake
index xxxxxxx..xxxxxxx 100644
--- a/vcpkg-port-config.cmake
+++ b/vcpkg-port-config.cmake
@@ -7,11 +7,7 @@ set(PYTHON3_VERSION_MINOR "@PYTHON_VERSION_MINOR@")
 set(PYTHON3_INCLUDE "include/python${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}")
 set(PYTHON3_HAS_EXTENSIONS "@PYTHON_HAS_EXTENSIONS@")
 set(site_base "")
-if(VCPKG_TARGET_IS_WINDOWS)
-  set(site_base "tools/python${PYTHON3_VERSION_MAJOR}/Lib")
-else()
-  set(site_base "lib/python${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}")
-endif()
+set(site_base "lib/python${PYTHON3_VERSION_MAJOR}.${PYTHON3_VERSION_MINOR}")
 set(PYTHON3_SITE "${site_base}/site-packages")
